src/braket/tasks/local_quantum_task.py

```diff
@@ -30,38 +44,332 @@ class LocalQuantumTask(QuantumTask):
+        """
+        return self._loop.create_task(self._async_run_internal())
+
+    def _run_event_loop(self, loop: AbstractEventLoop, task: Task) -> None:
+        """
+        Run the event loop with the given task.
+
+        Parameters:
+        - loop (asyncio.AbstractEventLoop): The asyncio event loop to use for running the task.
+        - task (coroutine): The coroutine or future
+                        that represents the task to run in the event loop.
+
+        Raises:
+        - Exception: If an exception occurs during the execution of the task.
+
+        Description:
+        This method sets the specified event loop using asyncio.set_event_loop(),
+        and then runs the event loop until the given task is completed.
+        If any other exception occurs during the execution of the task, it is re-raised
+        to be handled in the calling context.
+
+        Note:
+        - This method assumes that the event loop is not already running in the current thread.
+        - It is intended to be used internally or by advanced asyncio applications.
+
+        Example:
+        # Create and run an event loop with a task
+        loop = asyncio.get_event_loop()
+        task = loop.create_task(some_coroutine())
+        _run_event_loop(loop, task)
+        """
+        asyncio.set_event_loop(loop)
+        try:
+            loop.run_until_complete(task)
+        except Exception as e:
+            raise e
+
+    def _get_task(self) -> asyncio.Task:
+        if not hasattr(self, "_task") or (
+            self._task.done() and not self._task.cancelled() and self._result is None
+        ):
+            self._task = self._create_task()
+            self._thread = threading.Thread(
+                target=self._run_event_loop, args=(self._loop, self._task)
+            )
+            self._thread.start()
+
+        return self._task
+
+    async def _async_run_internal(self):
+        return self._run_internal(self._task_specification, *self._args, **self._kwargs)
+
+    @singledispatchmethod
+    def _convert_to_sim_format(
+        self,
+        task_specification: Union[
+            Circuit, Problem, Program, AnalogHamiltonianSimulation, AHSProgram
+        ],
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ) -> None:
+        raise NotImplementedError(f"Unsupported task type {type(task_specification)}")
+
+    @_convert_to_sim_format.register
+    def _(
+        self,
+        circuit: Circuit,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ):
+        simulator = self._delegate
+        if DeviceActionType.OPENQASM in simulator.properties.action:
+            validate_circuit_and_shots(circuit, shots)
+            program = circuit.to_ir(ir_type=IRType.OPENQASM)
+            program.inputs.update(inputs or {})
+            self._args = [program, shots] + list(self._args)
+            return
```

Comment:
Why are we doing this instead of just returning a value from the function?

Reply:
That is so true. I don't know what I was thinking when I did that, haha